### ISSUE TITLE
Add the ability to skip the mdleware on ajax requests on a specific route

### ISSUE THREAD

**Issue by agalazis** [i18next/i18next-express-middleware#57]
I know that you appreciate pull requests and I was tempted to just implement this and do pull request but then I thought  it would be better to discuss it in order to make sure I don't violate the philosophy of the middle-ware
One way is introducing igroreAjaxRoutes in the middle-ware configuration.
Another way would be to provide an ignoreRoute function:

```
 ignoreRoute:function(req,ignoreRoutes){
                       if (//do whatever checks you want){
                              return true;//to ignore
                      }
                      return false;//to use the route
}
```

 then use that function to check whether to skip and call next()

I guess the first is more clear but the second covers more cases.

**Comment by jamuhl** [comment]
like the idea ignoreRoute function. if array do what we do right now...if function call it with req and do next if true is returned.

PR is very welcome....;)

**Comment by isaachinman** [comment]
@jamuhl It's an old one, but is this on a roadmap?

**Comment by jamuhl** [comment]
@isaachinman there is no "roadmap" for the express middleware...PRs get merged on demand. If this is needed for the next.js implementation i can look into this the next days (it's a small thing to add).

**Comment by isaachinman** [comment]
@jamuhl I'm trying to match routes like this:

    /:lng/my-route

However, even with a whitelist in place, the express middleware is catching routes like:

    /_next/static/_app.js

And it's actually setting the `i18next` cookie value to `_next`. Not sure if this is an error on my part, or expected behaviour. Any tips?

**Comment by isaachinman** [comment]
I would have expected the whitelist to cause the middleware to ignore all `:lng` values that aren't whitelisted.

**Comment by isaachinman** [comment]
Not sure if this would be better solved by something like `ignoreRoute`, or just the ability to provide a custom `substring` to perform before doing lookups.

**Comment by jamuhl** [comment]
hm...cookie should be set like: https://github.com/i18next/i18next-express-middleware/blob/master/src/languageLookups/cookie.js#L31 

regarding the middleware getting triggered for statics - might be that handling changed from the road express 2 -> 4 --- so i will add that ignoreRoute function asap to handle such cases

**Comment by jamuhl** [comment]
published in i18next-express-middleware@1.5.0

`ignoreRoutes: function(req, res, options, i18next) { /* return true to ignore */ }`
